test/CadlRanchProjects/authentication/api-key/src/Authentication.ApiKey.csproj

```diff
@@ -10,6 +10,7 @@
 
   <ItemGroup>
     <Compile Include="$(AzureCoreSharedSources)AzureResourceProviderNamespaceAttribute.cs" LinkBase="Shared/Core" />
+    <Compile Include="$(AzureCoreSharedSources)AzureKeyCredentialPolicy.cs" LinkBase="Shared/Core" />
```

Comment:
what is the impact if this line was not added?  Seems we didn't get any problem of the compile error.

Reply:
Yes. It used to be added in `Directory.Build.targets` for autorest.csharp local test projects. so those test projects can compile successfully in autorest.csharp. But for those outside projects, without this, build will fail if the project has azure key authentication. To resolve this compile failure, we will add this to each sdk project file, and remove it from `Directory.Build.targets`.